routes/webauthn.go

```diff
@@ -0,0 +1,377 @@
+		CredentialName string `json:"credentialName,omitempty"`
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "failed to read request body", http.StatusBadRequest)
+		return
+	}
+
+	if len(body) > 0 {
+		if err := json.Unmarshal(body, &req); err != nil {
+			http.Error(w, invalidJSONMessage, http.StatusBadRequest)
+			return
+		}
+	}
+
+	options, sessionData, err := h.userSrvc.WebAuthnBeginRegistration(user, req.CredentialName)
+	if err != nil {
+		conf.Log().Request(r).Error("WebAuthn registration begin failed", "error", err)
+		http.Error(w, "failed to begin registration", http.StatusInternalServerError)
+		return
+	}
+
+	// Store session data in the response for the frontend to include in the finish request
+	response := map[string]interface{}{
+		"options":     options,
+		"sessionData": sessionData,
+	}
+
+	w.Header().Set(contentTypeHeader, applicationJSONType)
+	json.NewEncoder(w).Encode(response)
+}
+
+// PostRegisterFinish completes WebAuthn credential registration
+func (h *WebAuthnHandler) PostRegisterFinish(w http.ResponseWriter, r *http.Request) {
+	user := middlewares.GetPrincipal(r)
+	if user == nil {
+		conf.Log().Request(r).Error("WebAuthn registration finish - no user")
+		http.Error(w, "unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	conf.Log().Request(r).Info("WebAuthn registration finish started", "userID", user.ID)
+
+	// Read the raw body first for debugging
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "failed to read request body", http.StatusBadRequest)
+		return
+	}
+
+	var req struct {
+		SessionData interface{} `json:"sessionData"`
+		Response    interface{} `json:"response"`
+	}
+
+	if err := json.Unmarshal(body, &req); err != nil {
+		http.Error(w, invalidJSONMessage, http.StatusBadRequest)
+		return
+	}
+
+	conf.Log().Request(r).Info("WebAuthn registration finish - JSON decoded successfully")
+
+	// Convert response to JSON bytes for the service layer
+	responseBytes, err := json.Marshal(req.Response)
+	if err != nil {
+		conf.Log().Request(r).Error("WebAuthn registration finish - response marshal failed", "error", err)
+		http.Error(w, "invalid credential response", http.StatusBadRequest)
+		return
+	}
+
+	conf.Log().Request(r).Info("WebAuthn registration finish - calling service layer")
+
+	err = h.userSrvc.WebAuthnFinishRegistration(user, req.SessionData, responseBytes)
+	if err != nil {
+		conf.Log().Request(r).Error("WebAuthn registration finish failed", "error", err)
+		http.Error(w, "failed to finish registration", http.StatusBadRequest)
+		return
+	}
+
+	conf.Log().Request(r).Info("WebAuthn registration finish - success")
+
+	w.Header().Set(contentTypeHeader, applicationJSONType)
+	json.NewEncoder(w).Encode(map[string]interface{}{
+		"success": true,
+		"message": "WebAuthn credential registered successfully",
+	})
+}
+
+// PostLoginBegin starts WebAuthn authentication
+func (h *WebAuthnHandler) PostLoginBegin(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'm usually using [KeePassXC](https://keepassxc.org/) (with the Firefox plugin) for Passkey authentication on websites. If I try to register with the demo at [webauthn.io](https://webauthn.io/), my KeePassXC's passkey dialog is brought up as usual. However, with your implementation for Wakapi, I get a Firefox-native prompt instead, asking me to provide my biometrics. There's no option to use KeePassXC (or any other "software"-passkey).

Can you explain why that is? I'm not very familiar with the WebAuthn client APIs, yet.